treeherder/model/derived/artifacts.py

```diff
@@ -72,7 +72,7 @@ def get_job_artifact_list(self, offset, limit, conditions=None):
             debug_show=self.DEBUG,
         )
         for artifact in data:
-            artifact["blob"] = utils.decompress_if_needed(artifact["blob"])
+            artifact["blob"] = utils.decompress(artifact["blob"])
```

Comment:
Now that we don't need the try-except, I'd call `zlib.decompress()` directly here, and delete the function in utils.py, since the new `utils.decompress()` now doesn't do anything more than `zlib.decompress()`.

If you make that change, rebase and update the commit message to add a second line that explains we don't need the try-except since the old uncompressed data has now expired - I'll then merge this for you. Thanks! :-)